duckduckgo/iOS
# Autofill fix for edge case crash when tapping on search bar

Task/Issue URL: https://app.asana.com/0/414709148257752/1204144513391838/f
Tech Design URL:
CC:

**Description**:
Fixes an edge case where tapping on the search bar caused a crashed. 



**Steps to test this PR**:
The issue is that when tapping on the search bar, the viewModel changes the number of sections (since the first section `Save and Autofill Logins switch` is removed), but the tableView data was not reloaded. So when table scrolls up to focus on the search bar, `cellForRowAtIndexPath` is called to render a new row and since the tableView data and the viewModel data is different, `items[indexPath.row]` will crash if there is no item at that position 
1. Test on device (can't reproduce on simulator)
2. Import a very large number of logins (1k+)

3. Comment out the [one line code change](https://github.com/duckduckgo/iOS/pull/1617/files#diff-2bb09f3d45ff594d6804e6abbf6cebba30308d4e31b9c82544a732935f175382R638) (line 680), run up the app and hopefully you should experience the crash (it may take a few attempts of scrolling to different sections of the logins screen and tapping search again)
4. Add back in the code change
5. Test again and there should be no crash.
6. Perform some smoke tests (search, tap search results, cancel search etc) confirming there is no change in behaviour



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
